esp/services/ws_config2/ws_config2Session.hpp

```diff
@@ -0,0 +1,206 @@
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#ifndef _CONFIG2SERVICE_SESSION_HPP_
+#define _CONFIG2SERVICE_SESSION_HPP_
+
+
+#include "EnvironmentMgr.hpp"
+#include "build-config.h"
+#include <iterator>
+#include <algorithm>
+
+
+#define SESSION_KEY_LENGTH 10
+
+struct ConfigMgrSession {
+
+    // NOTE: all paths have NO trailing slash
+    std::string     username;            // owner of the session
+    std::string     schemaPath;          // path to schema files
+    std::string     sourcePath;          // path to environment files
+    std::string     activePath;          // path to active environment file
+    EnvironmentType configType;          // configuration type (XML, JSON, etc.)
+    std::string     masterConfigFile;    // master configuration file for the session
+    std::string     lockKey;             // key for write operations when session is locled
+    std::string     lastMsg;             // last error message
+    std::string     curEnvironmentFile;  // name of currentl loaded envronment file
+    bool            locked;              // true if locked
+    bool            modified;            // true if session has modified loaded environment
+    bool            externallyModified;  // true if another session modified the environment
+    EnvironmentMgr  *m_pEnvMgr;          // ptr to active environment manager for session
+
+
+    ConfigMgrSession() : locked(false), modified(false), m_pEnvMgr(nullptr), configType(UNDEFINED) { }
+    ~ConfigMgrSession()
+    {
+        if (m_pEnvMgr != nullptr)
+        {
+            delete m_pEnvMgr;
+            m_pEnvMgr = nullptr;
+        }
+    }
+
+
+    bool initializeSession(std::vector<std::string> &cfgParms)
+    {
+        bool rc = true;
+        m_pEnvMgr = getEnvironmentMgrInstance(configType);
+        if (m_pEnvMgr)
+        {
+           if (!m_pEnvMgr->loadSchema(schemaPath, masterConfigFile, cfgParms))
+           {
+               rc = false;
+               lastMsg = "Unable to load configuration schema, error = " + m_pEnvMgr->getLastSchemaMessage();
+           }
+        }
+        else
+        {
+            rc = false;
+            lastMsg = "Unrecognized configuration type";
+        }
+        return rc;
+    }
+
+
+    void getEnvironmentFullyQualifiedPath(const std::string &envFile, std::string &fullPath)
+    {
+        fullPath = sourcePath + "/" + envFile;
```

Comment:
There's an environment specific path separator in jlib.

Reply:
I replaced all occurrences of / with PATHSEPSTR